src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -1865,6 +1865,23 @@ public void setMaximumValidatorThreads(int number)
         validationExecutor.setMaximumPoolSize(number);
     }
 
+    public Map<String, Integer> getPendingTasksByKeyspaceAndTable()
+    {
+        Map<String, Integer> pendingCompactions = new HashMap<>();
+        for (String keyspaceName : Schema.instance.getKeyspaces())
+        {
+            for (ColumnFamilyStore cfs : Keyspace.open(keyspaceName).getColumnFamilyStores())
+            {
+                int estimatedRemainingTasks = cfs.getCompactionStrategy().getEstimatedRemainingTasks();
+                if (estimatedRemainingTasks > 0)
+                {
+                    pendingCompactions.put(String.format("%s/%s", keyspaceName, cfs.getColumnFamilyName()), estimatedRemainingTasks);
```

Comment:
It feels like that formatting should live in `CompactionStats` in this function should return a more generic Keyspace/Table key class.

Reply:
Nit: maybe worth also de-duping the logic with `CompactionMetrics` ?

Reply:
When the CFS object is sent over jmx a `NotSerializableException` is encountered. I'm looking into a workaround.

Reply:
The ideas:
- I could extend o.a.c.utils.Pair to implement Serializable, clean easy and beautiful but this opens up an object that is used in some of the most critical performance paths to abuse. There's probably a good reason why Pair isn't serializable.
- I could just return an array of strings of length 2 that represents the key and value, this is too unstructured.
- I can create a new wrapper class for the ks / cf pair and make it serializable. I'm going to go with this option, we can structure the ks/cf pairing in a generic object that will be re-usable but do so without modifying the existing Pair object.